### DISCUSSION TITLE
Provide a way to correct a single word when not in jinx-mode

### DISCUSSION THREAD

**Discussion by imarko** [minad/jinx#137] [Ideas]
In some situations I would like to use jinx to correct a single word in a buffer that is not in jinx-mode, similarly to how I would use `ispell-word` to do the same.

The closest thing I've found is `jinx-correct-word` but this has a couple of problems:
1. It puts the buffer into jinx-mode and leaves it on after it has finished. This is undesirable when I am not looking to see all spelling errors in the buffer, I only care about the current word.
2. It pops up the correction dialog even if the word is already correct. `ispell-word` just prints "someword is correct".

Currently I use the following clunky workaround, I am looking for a better way:

```
(defun my-jinx-correct-word ()
    (interactive)
    (let ((jinx-was-on jinx-mode))
      (unwind-protect
          (progn
            (unless jinx-was-on (jinx-mode 1))
            (let* ((bounds (jinx--bounds-of-word))
                   (word (buffer-substring (car bounds) (cdr bounds))))
              (if (jinx--word-valid-p (car bounds))
                  (message "%s is correct" word)
                (jinx-correct-word))))
        (unless jinx-was-on
          (jinx-mode -1)))))
```

**Comment by minad** [comment]
Hi!

> Currently I use the following clunky workaround, I am looking for a better way:

Why do you consider this command a clunky workaround? It seems like a valid solution for the problem at hand.

**Reply by imarko** [reply]
@minad Having to temporarily turn on jinx-mode to make the checker functions work is what I find clunky.

The function also has an issue when running on an already correct word that has punctuation after it. For example if i run the function with "jinx." in the buffer and the cursor after the period it doesn't recognize "jinx" as correct. It's probably a problem with the way my code tries to find the word bounds.
